dev/snippets/lib/main.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' hide Platform;
+
+import 'package:args/args.dart';
+import 'package:platform/platform.dart';
+
+import 'configuration.dart';
+import 'snippets.dart';
+
+/// Generates snippet dartdoc output for a given input, and creates any sample
+/// applications needed by the snippet.
+void main(List<String> argList) {
+  const Platform platform = LocalPlatform();
+  final Map<String, String> environment = platform.environment;
+  final ArgParser parser = ArgParser();
+  final List<String> snippetTypes =
+      SnippetType.values.map<String>((SnippetType type) => getEnumName(type)).toList();
+  parser.addOption(
+    'type',
+    defaultsTo: 'application',
+    allowed: snippetTypes,
+    allowedHelp: <String, String>{
+      getEnumName(SnippetType.application):
+          'Produce a code snippet complete with embedding the sample in an '
+          'application template.',
+      getEnumName(SnippetType.sample):
+          'Produce a nicely formatted piece of sample code. Does not embed the '
+          'sample into an application template.'
+    },
+    help: 'The type of snippet to produce.',
+  );
+  parser.addOption(
```

Comment:
Since you print out usage below, it'd be helpful to add `valueHelp` attributes to these